src/basic_recipes/voronoiplot.jl

```diff
@@ -0,0 +1,191 @@
+# For scatter-like inputs
+convert_arguments(::Type{<:Voronoiplot}, ps) = convert_arguments(PointBased(), ps)
+convert_arguments(::Type{<:Voronoiplot}, xs, ys) = convert_arguments(PointBased(), xs, ys)
+convert_arguments(::Type{<:Voronoiplot}, x::DelTri.VoronoiTessellation) = (x,)
+
+function plot!(p::Voronoiplot{<:Tuple{<:Vector{<:Point{N}}}}) where {N}
+    attr = copy(p.attributes)
+    smooth = pop!(attr, :smooth)
+
+    # from call pattern (::Vector, ::Vector, ::Matrix)
+    if N == 3
+        ps = map(ps -> Point2f.(ps), p[1])
+        attr[:color] = map(ps -> last.(ps), p[1])
+    else
+        ps = p[1]
+    end
+
+    # Handle transform_func early so tessellation is in cartesian space.
+    vorn = map(p.transformation.transform_func, ps, smooth) do tf, ps, smooth
+        transformed = Makie.apply_transform(tf, ps)
+
+        # TODO: Make this work with Point2f directly
+        M = Matrix{Float64}(undef, 2, length(transformed))
+        for (i, p) in enumerate(transformed)
+            M[:, i] .= p
+        end
+
+        tri = DelTri.triangulate(M)
+        vorn = DelTri.voronoi(tri)
+        if smooth
+            vorn = DelTri.centroidal_smooth(vorn)
+        end
+
+        return vorn
+    end
+
+    transform = Transformation(p.transformation; transform_func=identity)
+    return voronoiplot!(p, attr, vorn; transformation=transform)
+end
+
+function plot!(p::Voronoiplot{<:Tuple{<:DelTri.VoronoiTessellation}})
+    generators_2f = Observable(Point2f[])
+    PolyType = typeof(Polygon(Point2f[], [Point2f[]]))
+    polygons = Observable(PolyType[])
+
+    p.attributes[:_calculated_colors] = map(p.color, p[1]) do color, vorn
+        if color === automatic
+            # generate some consistent distinguishable colors
+            cs = [sum(DelTri.get_generator(vorn, i)) for i in DelTri.each_generator(vorn)]
+            reverse!(cs)
+        else
+            @assert(
+                length(color) == DelTri.num_points(DelTri.get_triangulation(vorn)),
+                "Color vector must have the same length as the number of generators, including any not yet in the tessellation."
+            )
+            [color[i] for i in DelTri.each_generator(vorn)] # this matches the polygon order
+        end
+    end
+
+    function update_plot(vorn)
+        bbox = map(p.unbounded_edge_extension_factor, p.bounding_box) do extent, bnd
+            isempty(DelTri.get_unbounded_polygons(vorn)) && return nothing
+            if bnd === automatic
+                # bb_ref = RefValue(Rect3f())
+                # foreach(DelTri.get_generators(vorn)) do (_, p)
+                #     update_boundingbox!(bb_ref, Point2f(p))
+                # end
+
+                # x0, y0, _ = minimum(bb_ref[]) .- extent .* widths(bb_ref[])
+                # x1, y1, _ = maximum(bb_ref[]) .+ extent .* widths(bb_ref[])
+
+                # return (x0, x1, y0, y1)
+                return DelTri.polygon_bounds(vorn, extent)
```

Comment:
I noticed that sometimes the poylgon vertices end up pretty far away from the input vertices:

```julia
using Random
Random.seed!(1234)

points = rand(Point2f, 50)
f, ax, p = voronoiplot(points)
f
```

![Screenshot from 2023-08-07 16-24-00](https://github.com/MakieOrg/Makie.jl/assets/10947937/08990003-e8e0-4901-ba8b-f2e2f70d6027)

I tried switching to a bounding box based on the input positions here to make this less of an issue, but DelaunayTriangulation isn't happy when triangle vertices end up outside the bounding box. Would it be possible to make that work, so we can avoid these far-away vertices here?

Reply:
Once I have merged https://github.com/DanielVandH/DelaunayTriangulation.jl/pull/76, we will be able to do some more testing here. Arbitrary (rectangular) bounding boxes will be able to be provided, and a better default for the bounding box will be 

```julia
DelTri.polygon_bounds(vorn, extent, include_polygon_vertices = false)
```

Reply:
The default behaviour is much nicer:

```julia
using Random
Random.seed!(1234)

points = rand(Point2f, 50)
f, ax, p = voronoiplot(points)
f
```

![1314ff41ec637f1c172055f6c0b378ca](https://github.com/MakieOrg/Makie.jl/assets/95613936/bb7fc98c-40bd-497e-9bcd-81fc99f38bf1)